workflow/setup_expt.py

```diff
@@ -50,18 +50,30 @@ def fill_EXPDIR(inputs):
 def update_configs(host, inputs):
 
     def _update_defaults(dict_in: dict) -> dict:
+        # Given an input dict_in of the form
+        # {defaults: {config_name: {var1: value1, ...}, }, config_name: {var1: value1, ...}}
+        # Replace values in ['defaults']['config_name']['var1'] with ['config_name']['var1']
+        # and return the ['defaults'] subdictionary as its own new dictionary.
         defaults = dict_in.pop('defaults', AttrDict())
```

Comment:
Defaults: now with recursion
```suggestion
        defaults = dict_in.pop('defaults', AttrDict())
        if 'defaults' in defaults:
            _update_defaults(defaults)
```